modules/GenericDataParser/src/server/generic_data_parser.R

```diff
@@ -3421,6 +3510,18 @@ runMain <- function(pathToGenericDataFormatExcelFile, reportFilePath=NULL,
       addError(paste0("Caught an error trying to validate dose response data. Please contact your administrator.: ", e$message))
     })
   }
+
+  # Validate Experiment Columns against Protocol Endpoints
+  # only proceed if it associated with an existing protocol (the protocol is not new)
+  if (!newProtocol) {
+    #if the protocol has strict endpoint matching enabled, proceed with validation
+    if (getProtocolStrictEndpointMatching(protocol) == TRUE) {
```

Comment:
We should do this validation either way. But the "strict" endpoint matching value should determine whether the validation throws errors (if strict) or warnings (if not strict).

It might make sense to pass the strictEndpointMatching value into `validateExperimentColumns` as an input variable.

However I think it would be a good idea to check the `client.protocol.endpointManager.enabled` config and skip this whole validation if that is set to false.